api/invite.go

```diff
@@ -1200,6 +1481,91 @@ func (a *Api) SendTeamInvite(res http.ResponseWriter, req *http.Request, vars ma
 
 }
 
+// @Summary Send invitation to a patient for monitoring purpose
+// @Description  create a notification for the invitee and send him an email with the invitation to be monitored. The patient account has to exist and the patient has to be a member of the team otherwise the invitation is rejected.
+// @ID hydrophone-api-SendMonitoringTeamInvite
+// @Accept  json
+// @Produce  json
+// @Param teamid path string true "Team ID"
+// @Param userid path string true "invited user id"
+// @Success 200 {object} models.Confirmation "invite details"
+// @Failure 400 {object} status.Status "teamId is not found"
+// @Failure 401 {object} status.Status "Authorization token is missing or does not provide sufficient privileges"
+// @Failure 403 {object} status.Status "Authorization token is invalid"
+// @Failure 409 {object} status.Status "user already has a pending or declined invite OR user is already part of the team"
+// @Failure 422 {object} status.Status "Error when sending the email (probably caused by the mailling service"
+// @Failure 500 {object} status.Status "Internal error while processing the invite, detailled error returned in the body"
+// @Router /send/team/monitoring/{teamid}/{userid} [post]
+// @security TidepoolAuth
+func (a *Api) SendMonitoringTeamInvite(res http.ResponseWriter, req *http.Request, vars map[string]string) {
+	tokenValue := req.Header.Get(TP_SESSION_TOKEN)
+	token := a.token(res, req)
+	if token == nil {
+		return
+	}
+
+	invitorID := token.UserId
+	if invitorID == "" {
+		res.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	patientid := vars["userid"]
+	teamid := vars["teamid"]
+
+	_, team, _ := a.getTeamForUser(tokenValue, teamid, token.UserId, res)
+
+	if team.ID == "" {
```

Comment:
I think we should also check that
1. The current user is an admin of the team (As only admins can dismiss invites...)
2. The current team is a monitored team